lib/functions.inc.php

```diff
@@ -95,6 +98,93 @@ function make_ad_password($password) {
     return $adpassword;
 }
 
+# Returns an array with "scheme" and "salt"
+function user_password_analyzer( $user_password_value ) {
+    $matches = array ();
+    if ( !preg_match('/{(\S+)}(\S+)/', $user_password_value, $matches) ) {
+        throw new InvalidArgumentException('Hashed password does not validate LDAP format');
```

Comment:
I do not plan to catch it, it is an assertion error. It should be catched by the exception_handler.
It will happen only the function is incorrectly used or the ldap server sends the pwdHistory in an incorrect format.